driver/src/main/java/org/neo4j/driver/internal/AbstractStatementRunner.java

```diff
@@ -21,19 +21,20 @@
 import java.util.Map;
 import java.util.concurrent.CompletionStage;
 
+import org.neo4j.driver.async.AsyncStatementRunner;
 import org.neo4j.driver.internal.types.InternalTypeSystem;
 import org.neo4j.driver.internal.util.Extract;
 import org.neo4j.driver.internal.value.MapValue;
 import org.neo4j.driver.Record;
 import org.neo4j.driver.Statement;
 import org.neo4j.driver.StatementResult;
-import org.neo4j.driver.StatementResultCursor;
+import org.neo4j.driver.async.StatementResultCursor;
 import org.neo4j.driver.StatementRunner;
 import org.neo4j.driver.Value;
 import org.neo4j.driver.Values;
 import org.neo4j.driver.types.TypeSystem;
 
-public abstract class AbstractStatementRunner implements StatementRunner
+public abstract class AbstractStatementRunner implements StatementRunner, AsyncStatementRunner
```

Comment:
I find this interesting: Bringing those two together again.
Where's the reactive statement runner in this context?

Reply:
`RxSession` takes a `NetworkSession` as input when creating. The `NetworkSession` indeed could implement `RxSession` too. But I feel it is not good to dump everything into one class. So the initial plan was to actually pull Blocking `Session` implementation out of `NetworkSession`. But as this PR only focus on making the API correct, so I did not split the implementation. Thus why this class implements two interfaces for example. Same for Session and Transaction.

Reply:
I also would suggest not use this internal class directly. If we split `NetworkSession` then we have two split this `AbstractStatementRunner` too.

Reply:
I don't plan. 

I will use `StatementRunner` and `ReactiveStatementRunner` interfaces separatly. 

The platform plumbing we are writing will provide one or the other then. 

The user is supposed to use one of our yet to write "Templates". Those are Spring Datas low level operations.

Based on that, there will be repositories.

In any case: The user says give my reactive or imperative and he gets a different template and should ideally never see the driver.